src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js

```diff
@@ -5,39 +5,55 @@ import 'whatwg-fetch';
 
 import './style.scss';
 
-function handleDownload(url, filename, accessToken) {
-  const headers = {};
+class DownloadButton extends React.Component {
+  constructor(props) {
+    super(props);
 
-  if (accessToken) {
-    headers.Authorization = `Bearer ${accessToken}`;
+    this.handleDownload = this.handleDownload.bind(this);
   }
 
-  fetch(url, {
-    method: 'GET',
-    headers,
-    responseType: 'blob'
-  }).then((response) => response.blob())
-    .then((blob) => {
-      const link = document.querySelector('.download-button__link');
-      link.href = window.URL.createObjectURL(blob);
-      link.click();
-    });
-}
+  handleDownload(url, filename, accessToken) {
+    const headers = {};
+
+    if (accessToken) {
+      headers.Authorization = `Bearer ${accessToken}`;
+    }
+
+    fetch(url, {
+      method: 'GET',
+      headers,
+      responseType: 'blob'
+    }).then((response) => response.blob())
+        .then((blob) => {
+          if (navigator.msSaveOrOpenBlob) {
+            navigator.msSaveOrOpenBlob(blob, filename);
+          } else {
+            const href = URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = href;
+            link.download = filename;
+            document.body.appendChild(link);
+            link.click();
+
+            window.URL.revokeObjectURL(href);
+            document.body.removeChild(link);
+          }
+        });
+  }
+
+  render() {
+    const { label, url, filename, accessToken } = this.props;
+    return (
+      <div className="download-button align-self-center">
+        <button
+          className="incident-detail__button"
```

Comment:
`<button>` Element has to have a `type` attribute with value 'button'

Reply:
gedaan